localstripe/resources.py

```diff
@@ -2350,6 +2350,27 @@ def __init__(self, id=None, name=None, type='service', active=True,
 
         schedule_webhook(Event('product.created', self))
 
+    @classmethod
+    def _api_list_all(cls, url, active=None, limit=None, starting_after=None,
+                      **kwargs):
+
+        active = try_convert_to_bool(active)
+        try:
+            if active is not None:
+                assert isinstance(active, bool)
+
+        except AssertionError:
+            raise UserError(400, 'Bad request')
+
+        li = super(Product, cls)._api_list_all(
+            url, limit=limit, starting_after=starting_after
+        )
+
+        if active is not None:
+            li._list = [obj for obj in li._list if obj.active == active]
+
+        return li
+
```

Comment:
Maybe you could add some unit testing?

It's inside the file `test.sh`, I propose:
```diff
--- a/test.sh
+++ b/test.sh
@@ -149,6 +149,14 @@ curl -sSfg -u $SK: $HOST/v1/products/PRODUCT1234
 
 curl -sSfg -u $SK: $HOST/v1/plans?expand[]=data.product
 
+count=$(curl -sSfg -u $SK: $HOST/v1/products?active=false \
+        | grep -oP 'total_count": \K([0-9]+)')
+[ "$count" -eq 0 ]
+
+count=$(curl -sSfg -u $SK: $HOST/v1/products?active=true \
+        | grep -oP 'total_count": \K([0-9]+)')
+[ "$count" -eq 9 ]
+
 code=$(curl -sg -o /dev/null -w '%{http_code}' -u $SK: \
             $HOST/v1/plans?expand[]=data.doesnotexist)
 [ "$code" -eq 400 ]
```

To run unit test you can launch localstripe in a terminal with `find -name '*.py' | entr -r python3 -m localstripe --from-scratch` then run `curl -X DELETE localhost:8420/_config/data && ./test.sh` in another terminal.

(same for the other commits)

Reply:
i added a few more tests. not complete, but at least basic coverage